source/grid/grid_tools.cc

```diff
@@ -3423,33 +3423,40 @@ namespace GridTools
       for (const unsigned int i : cell->vertex_indices())
         vertex_to_cell_map[cell->vertex_index(i)].insert(cell);
 
-    // Take care of hanging nodes
-    cell = triangulation.begin_active();
-    for (; cell != endc; ++cell)
+    if (triangulation.has_hanging_nodes())
       {
-        for (const unsigned int i : cell->face_indices())
+        Assert(triangulation.all_reference_cells_are_hyper_cube(),
+               ExcNotImplemented());
```

Comment:
And add a check that the hanging-node pass is only working for hyper-cube meshes at the moment.